public/app/percona/shared/services/user/User.service.ts

```diff
@@ -14,4 +14,13 @@ export const UserService = {
     );
     return is_platform_user;
   },
+  async getUserDetails(): Promise<UserDetailsResponse> {
+    const res: UserDetailsResponse = await api.get('/v1/user');
+    return res;
+  },
+  async setProductTourCompleted(completed: boolean): Promise<UserDetailsResponse> {
+    const payload: UserDetailsPutPayload = { product_tour_completed: completed };
+    const res: UserDetailsResponse = await api.put('/v1/user', payload);
+    return res;
```

Comment:
For both these calls, we can `return api.<get/put> ...`.
In the first case, no need for func body: ` => api.get(...)`

Reply:
Updated.

As for the screenshots, good catch! There was a 404 error visible, I've added disabling notifications for when the user details fetch fails. (If it fails, it will not show to tour)